### ISSUE TITLE
Properties are ignored

### ISSUE THREAD

**Issue by retailcoder** [x-vba/xdocgen#2]
`Function` and `Sub` members correctly generate `Procedure` objects, but `Property` members seem to be ignored for now.

Listing them as separate members would work, but it would be nice to structure `Property` procedures with the `Get` and `Let` (and/or `Set`) accessors under it (note they *may* not be next to each other in the source code), and then the `Property` procedure objects could have `ReadOnly` (and `WriteOnly`?) a flag.

Worth noting, to VBE the RHS/value parameter of a `Property Let` / `Property Set` procedure is always passed `ByVal`, regardless of what the parameter declaration says.

**Comment by x-vba** [comment]
Thanks for the info! Currently XDocGen doesn't support Properties and I haven't tested it too much with Class modules, so some functionality might be missing as well there. Hoping to add in Properties soon

**Comment by x-vba** [comment]
Pushed a quick update to include Property Let, Property Get, and Property Set. Fortunately Properties were close enough to the other Procedures that not too much additional work was needed on the regex. It's a little experimental since I only did some basic testing, but hopefully will work as expected similar to Functions and Subs. One note is that Property names include either (Get), (Let), or (Set) at the end of them to account for the same names for the getters and setters. Website should update in a bit to reflect the new code.

**Comment by x-vba** [comment]
Actually just realized the second point that they default as ByVal, I'll keep this open for now until I get to that too

**Comment by retailcoder** [comment]
Yeah it's... I don't know if it's best to just report whatever the code *appears* to say (ByRef, implicit or not), or what VBA actually does; note that this is only for the RHS/value parameter, which would be the last parameter of a `Property Let/Set` procedure.
Basically a `ByRef` modifier on that parameter is just outright ignored, it's always `ByVal` (found out reading language specs, confirmed implemented as such in VBA7 using `VarPtr`; if the argument was passed by reference, `VarPtr` would yield the same address both at the call site and within the procedure, as it does with a `ByRef` parameter in any `Sub` procedure).

That's potentially confusing, but also a learning opportunity and a potential explanation for unexpected behavior in user code.
